x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/structurefinder/TextStructure.java

```diff
@@ -254,6 +270,9 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         out.writeOptionalBoolean(shouldTrimFields);
         out.writeOptionalString(grokPattern);
+        if (out.getVersion().onOrAfter(Version.V_8_5_0)) {
+            out.writeOptionalString(ecsCompatibility);
```

Comment:
If you remove all the loopholes that permit `null` then this doesn't need to be optional:

```suggestion
            out.writeString(ecsCompatibility);
```